snappy-tools/src/main/scala/org/apache/spark/sql/columntable/ColumnFormatRelation.scala

```diff
@@ -158,21 +159,38 @@ class ColumnFormatRelation(
   }
 
   override def uuidBatchAggregate(accumulated: ArrayBuffer[UUIDRegionKey],
-      batch: CachedBatch): ArrayBuffer[UUIDRegionKey] = {
+      batch: CachedBatch, index: Int): ArrayBuffer[UUIDRegionKey] = {
+
     //TODO - currently using the length from the part Object but it needs to be handled more generically
     //in order to replace UUID
     // if number of rows are greater than columnBatchSize then store otherwise store locally
-    if (batch.numRows >= columnBatchSize) {
-      //TODO - rddID should be passed to the executors so that cachedBatch size can be shown be correctly even for non embedded mode
+    //TODO - rddID should be passed to the executors so that cachedBatch size can be shown be correctly even for non embedded mode
+
+    //Change for preserve partition : {
+    // We are inserting a cached batch to column table directly
+    // in case of preserve partition. We are not considering small batches.
+    // We will be simply putting the data as CachedBatches. This will
+    // ensure the query results will be ok even though the storage is suboptimal.
+    // Once we come up with a cleaner solution we can start inserting
+    // residual rows to row buffer
```

Comment:
Why does it matter whether row goes to row buffer or directly? All we need to ensure is that the bucketId column (which is used for partitioning in row buffer) is the same as partition ID. I think it should be quite simple to add row buffer support which will be important especially for streaming cases (the most important use-case for preservePartitioning) where lots of small batches will be the norm rather than exception.